### ISSUE TITLE
Battle Royal visualization is broken

### ISSUE THREAD

**Issue by Datseris** [JuliaDynamics/AgentsExampleZoo.jl#23]
agent color doesn't change, to say the least. Probably we haven't updated the example to v5+.

**Comment by BonelessPi** [comment]
Newcomer to Agents.jl here. I attempted to follow the guide online and noticed that it didn't work as intended. After a little debugging, it seems that all the walk! calls need to be passed ifempty=false, then things seemed to be working better. I assume the default behavior changed from false to true but that's just a guess.

**Comment by Datseris** [comment]
Thanks @BonelessPi , do you mind putting a pull request together?

**Comment by Datseris** [comment]
yeah some of the examples here were written 2-3 major versions of Agents.jl in the past. For most this didn't matter, but for some it did.
